### PR TITLE
PLU-81: [IF-THEN-9002] Show special messages when testing branch

### PR DESCRIPTION
**Pull request by ogp-weeloong**
## Problem
Our messaging around test steps that would have been skipped if published is confusing.

## Solution
We will remove that messaging, and instead clearly explain if a branch is taken or not when a user tests the branch.

This implies the following engineering changes:
1. Remove the `skippedIfPublished` graphql field and associated backend code
2. Add an edge case to `TestSubstep` to show if-then messages instead of variables.
   * _This results in 2 edge cases in the UI for if-then: in `ChooseAppAndEventSubstep` and `TestSubstep`_
3. Upgrade design system to 1.8.0 to use the new success infobox from @myself

## Tests
- Check that new messaging shows up in taken and non-taken branches
- Check that testing actions within not-taken branches will show the default variables list [Screenshots - [taken branch](https://github.com/opengovsg/plumber/assets/135598754/1f561dff-3492-4f59-9da5-8fc2597e296c), [not taken branch](https://github.com/opengovsg/plumber/assets/135598754/647b542a-a96a-46a8-8edb-7ea0b27cd8a4)]
- Regression test (for DS upgrade): check that I can still manipulate and create pipes
- Regression test: check that the variable list still shows up correctly in non-if-then steps [Screenshots -  [step with instructions](https://github.com/opengovsg/plumber/assets/135598754/333c8c9c-1fca-4394-b563-8f127bcfc284), [step with instructions and data not found](https://github.com/opengovsg/plumber/assets/135598754/4b2894ff-ed0e-456d-baaf-f874a59afacc), [step without instructions](https://github.com/opengovsg/plumber/assets/135598754/cebab6d3-c56b-4352-adfe-8a46a7ba859a), [step without instructions and data not found](https://github.com/opengovsg/plumber/assets/135598754/48020908-b8dd-4bdc-bb0e-5d798176ecd5)]

## Deploy Notes
**Updated dependencies**:
- `@opengovsg/design-system-react`: Bump to v1.6.0 to use success infobox. Not bumping to v1.8.0 because it breaks build

### CODE REVIEW

**Comment by pregnantboy** [comment]
we need to discuss about the copy with the designers perhaps. lgtm otherwise

**Comment by ogp-weeloong** [comment]
Discussed with PD, we will make the following changes:
* Revert warning infobox
* Edge case the branch step, update with specialised copy on whether branch passed or not

File: packages/frontend/src/components/TestSubstep/TestResult.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+export default function TestResult(props: TestResultsProps): JSX.Element {
+  const { step, selectedActionOrTrigger, stepsWithVariables, isExecuted } =
+    props
+
+  // No data only happens if user hasn't executed yet, or step returned null.
+  if (stepsWithVariables.length == 0) {
+    if (isExecuted) {
+      return (
+        <Infobox variant="warning" width="full">
```
Comment (m0nggh) [inline packages/frontend/src/components/TestSubstep/TestResult.tsx:38]: nit: need to align colour with the other warning colors: should be yellow.200

File: packages/frontend/src/components/TestSubstep/TestResult.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        <Infobox variant="success" w="full">
+          <Text>
+            Based on your sample data, it meets the conditions that you have set
+            up and your pipe <Text as="b">would have</Text> continued.
+          </Text>
+        </Infobox>
+      )
+    } else {
+      return (
+        <Infobox variant="warning" w="full">
```
Comment (m0nggh) [inline packages/frontend/src/components/TestSubstep/TestResult.tsx:83]: same thing here!

File: packages/frontend/src/theme/foundations/colors.ts [hunk 3]
```diff
@@ -2,6 +2,9 @@
 // size.
 // Imported from BbBear generated on Fri, 23 Jun 2023 09:14:33 GMT
 const bbBearColors = {
+  grey: {
+    '900': '#2C2E34',
+  },
```
Comment (m0nggh) [inline packages/frontend/src/theme/foundations/colors.ts:7]: nit: is this used in another pr branch? cus i dont see this in the code haha

File: packages/frontend/src/components/TestSubstep/TestResult.tsx (hunk 1 above)
Comment (ogp-weeloong) [reply packages/frontend/src/components/TestSubstep/TestResult.tsx:38]: thanks, have fixed this in next PR #237

File: packages/frontend/src/theme/foundations/colors.ts (hunk 3 above)
Comment (ogp-weeloong) [reply packages/frontend/src/theme/foundations/colors.ts:7]: wah good catch. I think i imported by accident.

**Review by m0nggh — APPROVED** [review]
lgtm after the minor nits! good jobb
